paymentsheet/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -224,24 +220,43 @@ internal abstract class BaseAddPaymentMethodFragment(
         @VisibleForTesting
         internal fun getFormArguments(
             hasCustomer: Boolean,
-            saveForFutureUse: Boolean,
             supportedPaymentMethodName: String,
+            stripeIntent: StripeIntent,
             merchantName: String,
             amount: Amount? = null,
             billingAddress: PaymentSheet.BillingDetails? = null
         ): FormFragmentArguments {
+            // Has effect of setting off session on PIs (not SIs) and also impacts reopening
+            // the card
             var saveForFutureUseValue = true
+            // This will impact the setting of the off_session on confirm
             var saveForFutureUseVisible = true
-            if (!hasCustomer) {
-                saveForFutureUseValue = false
-                saveForFutureUseVisible = false
-            }
 
-            // The order is important here, even if there is a customer the save for future
-            // use value should be true to collect all the details
-            if (saveForFutureUse) {
+            val supportedPaymentMethod =
+                SupportedPaymentMethod.fromCode(supportedPaymentMethodName)
+
+            val isSetupIntent = stripeIntent is SetupIntent
+            val isPaymentIntentOffSession = (stripeIntent as? PaymentIntent)
+                ?.setupFutureUsage == StripeIntent.Usage.OffSession
+            if (isSetupIntent || isPaymentIntentOffSession) {
                 saveForFutureUseVisible = false
                 saveForFutureUseValue = true
```

Comment:
I'm not sure what this controls but TBC we should not set `setup_future_usage` on `/confirm` if we aren't showing the checkbox.

Reply:
I think this is an issue because it is only new payment methods that this impacts, I will create a follow up PR, in an effort to keep the commits manageable.

Reply:
ok, just tested it on master, and it isn't setting the flag on confirm unless the checkbox is set AND visible by the user, so no bug, but I will follow up with some renames in another PR.